receipt/src/androidTest/java/com/hyperwallet/android/receipt/ListReceiptsTest.java

```diff
@@ -0,0 +1,265 @@
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(0,
+                hasDescendant(withText(com.hyperwallet.android.receipt.R.string.debit)))));
+        onView(withId(R.id.list_receipts)).check(
+                matches(atPosition(0, hasDescendant(withText(R.string.transfer_to_prepaid_card)))));
+        onView(withId(R.id.list_receipts)).check(
+                matches(atPosition(0, hasDescendant(withText("- 18.05")))));
+        onView(withId(R.id.list_receipts)).check(
+                matches(atPosition(0, hasDescendant(withText("May 02, 2019")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(0, hasDescendant(withText("USD")))));
+
+        onView(withId(R.id.list_receipts)).check(new RecyclerViewCountAssertion(1));
+    }
+
+    @Test
+    public void testListReceipts_displayUnknownTransactionType() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_OK).withBody(sResourceManager
+                .getResourceContent("receipt_unknown_type_response.json")).mock();
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_NO_CONTENT).withBody("").mock();
+
+        // run test
+        mActivityTestRule.launchActivity(null);
+
+        // assert
+        onView(allOf(instanceOf(TextView.class), withParent(withId(R.id.toolbar))))
+                .check(matches(withText(R.string.title_activity_receipt_list)));
+        onView(withId(R.id.list_receipts)).check(matches(isDisplayed()));
+
+        onView(withId(R.id.list_receipts))
+                .check(matches(atPosition(0, hasDescendant(withText("June 2019")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(0,
+                hasDescendant(withText(com.hyperwallet.android.receipt.R.string.credit)))));
+        onView(withId(R.id.list_receipts)).check(
+                matches(atPosition(0, hasDescendant(withText(R.string.unknown_type)))));
+        onView(withId(R.id.list_receipts)).check(
+                matches(atPosition(0, hasDescendant(withText("+ 25.00")))));
+        onView(withId(R.id.list_receipts)).check(
+                matches(atPosition(0, hasDescendant(withText("June 02, 2019")))));
+        onView(withId(R.id.list_receipts)).check(matches(atPosition(0, hasDescendant(withText("CAD")))));
+
+        onView(withId(R.id.list_receipts)).check(new RecyclerViewCountAssertion(1));
+    }
+
+    @Test
+    public void testListReceipt_userHasNoTransactions() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_NO_CONTENT).withBody("").mock();
+
+        // run test
+        mActivityTestRule.launchActivity(null);
+
+        onView(withId(R.id.toolbar)).check(matches(isDisplayed()));
+        onView(allOf(instanceOf(TextView.class), withParent(withId(R.id.toolbar))))
+                .check(matches(withText(R.string.title_activity_receipt_list)));
+        //todo: check empty view when it will be ready
```

Comment:
@skoong who's addressing this todo?

Reply:
@fmattos-hw this todo is addressed to @peter-joseph - if I'm not mistaken - there should be a view for empty list (like label: you don't have transactions... etc.) in the Fragment